### PR TITLE
Fix scripts not exiting properly in CI; refactor

### PR DESCRIPTION
**Pull request by starsirius**
https://artsy.slack.com/archives/C02BC3HEJ/p1529518134000475

screen shot 2018-06-21 at 11 49 37 am

We noticed in a [deployment](https://circleci.com/gh/artsy/force/6825) build, the `bucket-assets` command failed but the the build didn't exit immediately and didn't fail as expected. The reason was we run the scripts with `bash` but not as an executable and the script header (`-e` in this case) was ignored. I refactored all the scripts I found and run them as executables, and also fixed a few issues I noticed.

I tried to commit and document incremental changes so it might be easier to follow them one by one.

I think @saolsen worked on the scripts recently but feel free to re-assign to someone else. Thanks!

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ else
   git push --force heroku $CIRCLE_SHA1:refs/heads/master
 fi
 
-if [ "$DEPLOY_ENV" = 'production' ]
+if [ "$DEPLOY_ENV" = 'production' ]; then
```
Comment (starsirius) [inline scripts/deploy.sh]: Lacking a `then` here caused a syntax error but the [build](https://circleci.com/gh/artsy/force/6829) was still 💚 .

screen shot 2018-06-21 at 1 54 36 am

Comment (joeyAghion) [reply scripts/deploy.sh]: Nice catch

File: package.json [hunk 2]
```diff
@@ -7,21 +7,21 @@
...
     "prepare": "patch-package",
     "prettier-write": "yarn run prettier --write",
     "relay": "node scripts/relayCompiler.js",
-    "sentry": "sh scripts/sentry.sh",
-    "start": "sh scripts/start.sh",
+    "sentry": "scripts/sentry.sh",
+    "start": "scripts/start.sh",
     "start:relay": "yarn relay && concurrently --raw --kill-others 'yarn relay --watch' 'yarn start'",
     "sync-schema": "rm -rf data && mkdir data && graphql-fetch-schema https://metaphysics-staging.artsy.net -o data",
-    "test": "sh scripts/test.sh",
```
Comment (joeyAghion) [inline package.json]: Do these updates mean the `.sh` files must be executable, or does `yarn` work either way?

Comment (starsirius) [reply package.json]: I think so. `yarn` would just run whatever command here, so I also `chmod +x` all the scripts in this PR.

**Comment by starsirius** [comment]
I mistakenly used "shebang" to refer to script header in the commit messages so I amended them to be more accurate.

**Comment by saolsen** [comment]
Cool, this looks great.

**Pull request merged**